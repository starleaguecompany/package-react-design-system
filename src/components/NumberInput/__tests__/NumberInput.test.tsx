import * as React from 'react'
import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Menu } from '@starleaguecompany/react-icons'

import { NumberInput } from '../../..'

describe('NumberInput', () => {
  test('should render correctly with no props', () => {
    render(<NumberInput />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('NumberInput')).toHaveClass('container')
  })

  test('should render correctly with attributes', () => {
    render(<NumberInput id="test-id" className="test-class" />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('NumberInput')).toHaveClass('test-class')
    expect(screen.getByRole('textbox')).toHaveAttribute('id', 'test-id')
  })

  test('should render correctly with value prop', () => {
    render(<NumberInput value={10} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveValue('10')
    expect(screen.getByText('10')).toBeVisible
  })

  test('should render correctly with defaultValue prop', () => {
    render(<NumberInput defaultValue={10} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveValue('10')
    expect(screen.getByText('10')).toBeVisible
  })

  test('should render correctly with postfix prop', () => {
    render(<NumberInput value={10} postfix="₽" />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('₽')).toBeVisible()
    expect(screen.getByText('₽')).toHaveClass('inputPostfix')
    expect(screen.getByTestId('NumberInput')).toHaveTextContent('10 ₽')
  })

  test('should render correctly with disabled prop', () => {
    const { container } = render(<NumberInput disabled />)
    const disabledElements = container.querySelectorAll('.disabled')

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('TextInput')).toHaveClass('disabled')
    expect(screen.getByRole('textbox')).toBeDisabled()
    expect(screen.getByRole('textbox')).toHaveClass('disabled')
    expect(disabledElements.length).toEqual(2)
  })

  test('should render correctly with invalid prop', () => {
    const { container } = render(<NumberInput invalid />)
    const invalidElements = container.querySelectorAll('.invalid')

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('TextInput')).toHaveClass('invalid')
    expect(screen.getByRole('textbox')).toHaveClass('invalid')
    expect(invalidElements.length).toEqual(2)
  })

  test('should render correctly with format prop', () => {
    render(<NumberInput value={31122021} format="##/##/####" />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('31/12/2021')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('31/12/2021')
  })

  test('should render correctly with readOnly prop', () => {
    render(<NumberInput value={50} readOnly />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveAttribute('readonly')
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('[Backspace]')
    expect(screen.getByRole('textbox')).toHaveValue('50')
  })

  test('should render correctly with max prop', () => {
    render(<NumberInput max={100} value={150} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('100')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('100')
  })

  test('should render correctly with min prop', () => {
    render(<NumberInput min={100} value={50} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('100')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('100')
  })

  test('should render correctly with step prop (step < 1)', () => {
    render(<NumberInput step={0.1} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('12.123')
    expect(screen.getByText('12,123')).toBeVisible()
    userEvent.click(document.body)
    expect(screen.getByText('12,1')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('12,1')
  })

  test('should render correctly with step prop (step > 1)', () => {
    render(<NumberInput step={10} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('128')
    userEvent.click(document.body)
    expect(screen.getByText('130')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('130')
  })

  test('should render correctly with decimalSeparator prop', () => {
    render(<NumberInput decimalSeparator="," />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('10.123')
    userEvent.click(document.body)
    expect(screen.getByText('10,123')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('10,123')
  })

  test('should allow to input only digits, comma and dot', () => {
    render(<NumberInput value={10} min={0} max={100} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('abc ,.5')
    expect(screen.getByText('10,5')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('10,5')
  })

  test('should render correctly on update value', () => {
    const onChangeCallback = jest.fn()
    const { rerender } = render(<NumberInput value={50} decimalSeparator="," onChange={onChangeCallback} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    rerender(<NumberInput value={150.1} decimalSeparator="," onChange={onChangeCallback} />)
    expect(screen.getByText('150,1')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('150,1')
    expect(onChangeCallback).toBeCalledTimes(0)
  })

  test('should execute onChange callback on input value', () => {
    const onChangeCallback = jest.fn(value => value)
    render(<NumberInput onChange={onChangeCallback} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('1,23')
    expect(onChangeCallback).toBeCalledTimes(3)
    expect(onChangeCallback).toReturnWith(1)
    expect(onChangeCallback).toReturnWith(1.2)
    expect(onChangeCallback).toReturnWith(1.23)
  })

  test('should render correctly with undefined value', () => {
    const onChangeCallback = jest.fn(value => value)
    render(<NumberInput onChange={onChangeCallback} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByRole('textbox')).toHaveValue('')
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('1')
    userEvent.keyboard('[Backspace]')
    expect(onChangeCallback).toBeCalledTimes(2)
    expect(onChangeCallback).toReturnWith(1)
    expect(onChangeCallback).toReturnWith(undefined)
  })

  test('should render correctly with loading prop', () => {
    const { rerender } = render(<NumberInput loading />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('NumberInput')).toContainElement(screen.getByTestId('Spinner'))
    rerender(<NumberInput />)
    expect(screen.getByTestId('NumberInput')).not.toContainElement(screen.queryByTestId('Spinner'))
  })

  test('should render correctly with icon prop', () => {
    render(<NumberInput icon={<Menu />} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('NumberInput')).toContainElement(screen.getByTestId('Icon'))
    expect(screen.getByTestId('Icon')).toContainElement(screen.getByTestId('Menu'))
  })

  test('should render correctly with autoFocus prop', () => {
    render(<NumberInput autoFocus />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByTestId('TextInput')).toHaveClass('focused')
    expect(screen.getByRole('textbox')).toHaveFocus()
  })

  test('should render correctly with thousandSeparator prop', () => {
    render(<NumberInput value={123456789} thousandSeparator=" " />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('123 456 789')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('123 456 789')
  })

  test('should render correctly with decimalScale prop', () => {
    render(<NumberInput value={12.34444} decimalScale={2} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('12,34')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('12,34')
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard('[Backspace]')
    expect(screen.getByText('12,30')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('12,30')
  })

  test('should render correctly with fixedDecimalScale prop', () => {
    render(<NumberInput value={100} decimalScale={1} fixedDecimalScale={false} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('100')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('100')
    userEvent.click(screen.getByRole('textbox'))
    userEvent.keyboard(',12345')
    expect(screen.getByText('100,1')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('100,1')
    userEvent.keyboard('[Backspace]')
    userEvent.keyboard('[Backspace]')
    expect(screen.getByText('100')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('100')
  })

  test('should format input value on blur', async () => {
    const onChangeCallback = jest.fn(value => value)
    render(<NumberInput min={0} max={100} onChange={onChangeCallback} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.click(screen.getByRole('textbox'))
    userEvent.type(screen.getByRole('textbox'), '200')
    userEvent.click(document.body)
    await waitFor(() => expect(onChangeCallback).toBeCalledTimes(4))
    await waitFor(() => expect(onChangeCallback).toHaveLastReturnedWith(100))
  })

  test('should format value on enter key press', async () => {
    const onKeyPressCallback = jest.fn(v => v)
    render(<NumberInput onKeyPress={onKeyPressCallback} max={100} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    userEvent.keyboard('[Enter]')
    expect(onKeyPressCallback).toBeCalledTimes(0)
    await userEvent.type(screen.getByRole('textbox'), '200{enter}', { delay: 5 })
    //TODO: fix when testing-library/user-event is updated
    //https://github.com/testing-library/user-event/issues/766
    // await waitFor(() => expect(onKeyPressCallback).toBeCalledTimes(4))
    await waitFor(() => expect(screen.getByRole('textbox')).toHaveValue('100'))
  })

  test('should reset value', async () => {
    const { rerender } = render(<NumberInput value={500} />)

    expect(screen.getByTestId('NumberInput')).toMatchSnapshot()
    expect(screen.getByText('500')).toBeVisible()
    expect(screen.getByRole('textbox')).toHaveValue('500')

    rerender(<NumberInput value={undefined} />)
    await waitFor(() => expect(screen.getByRole('textbox')).toHaveValue(''))
  })
})
