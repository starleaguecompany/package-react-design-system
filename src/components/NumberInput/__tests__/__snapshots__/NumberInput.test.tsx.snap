// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput should allow to input only digits, comma and dot 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          10
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should execute onChange callback on input value 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should format input value on blur 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should format value on enter key press 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly on update value 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          50
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="50"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with attributes 1`] = `
<div
  class="test-class container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          id="test-id"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with autoFocus prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container focused active"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input active"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with decimalScale prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          12,34
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="12,34"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with decimalSeparator prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with defaultValue prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          10
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with disabled prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container disabled"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input disabled"
          disabled=""
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with fixedDecimalScale prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          100
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with format prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          31/12/2021
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="31/12/2021"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with icon prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container with-icon"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <span
      class="icon container size-20 shape-round"
      data-qa="Icon"
    >
      <svg
        data-qa="Menu"
        fill="none"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4 5h12M4 10h12M4 15h12"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`NumberInput should render correctly with invalid prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container invalid"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input invalid"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with loading prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container with-icon"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
    <span
      class="icon container size-20"
      data-qa="Spinner"
    >
      <svg
        class="svg"
        viewBox="0 0 150 150"
      >
        <circle
          class="circle circle-1"
          cx="75"
          cy="75"
          r="60"
        />
        <circle
          class="circle circle-2"
          cx="75"
          cy="75"
          r="60"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`NumberInput should render correctly with max prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          100
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with min prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          100
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="100"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with no props 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with postfix prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          10
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="10"
        />
        <div
          class="inputPostfix"
        >
           
          ₽
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with readOnly prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          50
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          readonly=""
          spellcheck="false"
          type="text"
          value="50"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with step prop (step < 1) 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with step prop (step > 1) 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with thousandSeparator prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          123 456 789
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="123 456 789"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with undefined value 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        />
        <input
          autocomplete="off"
          class="input"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should render correctly with value prop 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          10
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="10"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`NumberInput should reset value 1`] = `
<div
  class="container"
  data-qa="NumberInput"
>
  <div
    class="container"
    data-qa="TextInput"
  >
    <div
      class="inputSegment centered"
    >
      <div
        class="inputWrapper"
      >
        <div
          class="inputValue"
        >
          500
        </div>
        <input
          autocomplete="off"
          class="input filled"
          inputmode="decimal"
          spellcheck="false"
          type="text"
          value="500"
        />
      </div>
    </div>
  </div>
</div>
`;
